examples/platforms/cc2538/CMakeLists.txt

```diff
@@ -28,13 +28,19 @@
 
 set(OT_PLATFORM_LIB "openthread-cc2538" PARENT_SCOPE)
 
+if(OT_CONFIG MATCHES "none")
+    set(OT_CONFIG "openthread-core-cc2538-config.h")
+    set(OT_CONFIG ${OT_CONFIG} PARENT_SCOPE)
+endif()
+
 list(APPEND OT_PLATFORM_DEFINES
-    "OPENTHREAD_PROJECT_CORE_CONFIG_FILE=\"openthread-core-cc2538-config.h\""
     "OPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE=\"openthread-core-cc2538-config-check.h\""
     "OPENTHREAD_CONFIG_NCP_UART_ENABLE=1"
 )
 set(OT_PLATFORM_DEFINES ${OT_PLATFORM_DEFINES} PARENT_SCOPE)
 
+list(APPEND OT_PLATFORM_DEFINES "OPENTHREAD_PROJECT_CORE_CONFIG_FILE=\"${OT_CONFIG}\"")
```

Comment:
Nit: Seems this isn't needed, as the root CMakeLists will set the OT_PRIVATE_DEFINES to include the config file using the OT_CONFIG that was just set in the parent scope.

Reply:
Not sure if I follow. Here we are setting `OT_PLATFORM_DEFINES` but in root we set a different variable `OT_PRIVATE_DEFINES`.

Reply:
Ah, I saw the OT_PLATFORM_DEFINES added to the OT_PRIVATE_DEFINES in the higher scope, so made this comment because there is then a duplicate entry in the OT_PRIVATE_DEFINES list. I forgot the core config file was required for the platform layer, so not an issue.